### ISSUE TITLE
Trial 44 failed because of the following error: ValueError('All layers added to a Sequential model should have unique names.

### ISSUE THREAD

**Issue by ali3assi** [optuna/optuna#2350]
when running the paralisation in optuna as follows:
 ```
study = optuna.create_study(
        direction="maximize",
        study_name="test", 
        storage="sqlite:///mydb.db", 
        load_if_exists= True, #to skip creating a new study if it already exists. 
        pruner=HyperbandPruner(min_resource=1, max_resource=100, reduction_factor=3),
    )
```

i get the following exception:
`[W 2021-02-15 18:53:35,171] Trial 44 failed because of the following error: ValueError('All layers added to a Sequential model should have unique names. Name "batch_normalization" is already the name of a layer in this model. Update the `name` argument to pass a unique name.')

  

## Environment

- Optuna version:
- Python version:
- OS:
- (Optional) Other libraries and their versio

```
# error messages, stack traces, or logs
```

## Steps to reproduce

1.
2.
3.

## Reproducible examples (optional)

```python
# python code
```

**Comment by crcrpar** [comment]
Could you share the (minimal) reproducible code?

**Comment by ali3assi** [comment]
@crcrpar thanks for your reply. I added for each layer a name attribute so the above exception is not thrown. BUT I have something strange since it is expected to execute my code in parallel way and unfortunately I cannot see the parallelization like:
```
Trial 1: ...
Trial 10:....
```
So the execute in normal way. 
1- In fact I have the code in file test.py which include all optuna stuff like study.optimize etc.
and in another file I have the following:
```
import multiprocessing as mp
from concurrent.futures import ProcessPoolExecutor
import os
from multiprocessing import Pool

def run_process (process):
    os.system('python {}'.format(process))
if __name__ == "__main__":
    processes = ['C:/Users/ali/OneDrive/Desktop/pythonProjects/Deep-learning-algo-master/src/TestDriver.py']*2
    pool = Pool()
    pool.map(run_process, processes)
    pool.close    
    
```
2- I tried to log all to neptune, and the code create two experiments in Neptune which incorrect. Usually, and logically, I should have the same experiment in neptune dahsborad.

**Comment by nzw0301** [comment]
@ali3assi Could you share the codes related to both sampling hyper-parameters with optuna and model construction?
This is because your error looks like a deep learning framework's error.

**Comment by ali3assi** [comment]
@nzw0301  Unfortunately I cannot since the code is a little bit big. The error is generated only when I add n_jobs in optimize function of optuna.

**Comment by nzw0301** [comment]
@ali3assi okay, could you check this related issue https://github.com/optuna/optuna/issues/1885 ?

**Comment by ali3assi** [comment]
@nzw0301 Thanks I will check it but can you please tell me how can run my code in parallel using the code and not the terminals

**Comment by ali3assi** [comment]
> @ali3assi okay, could you check this related issue #1885 ?

I added the name for each layer, but not working

**Comment by nzw0301** [comment]
@ali3assi I'm afraid that we cannot tell you that without full reproducible codes :(

**Comment by ali3assi** [comment]
@nzw0301 Thanks just I fixed the problem following the same way. I have one layer without name, so after adding it the name, the code work fine. BUT the parallelisation still not working and the code work as it is without parallelisation.

**Comment by nzw0301** [comment]
@ali3assi  Awesome! I'm glad to hear that. I think that you report the parallelisation issue as another issue.

**Comment by crcrpar** [comment]
Let me close this issue because the main problem seems to have been resolved :)
